### PR TITLE
60 add device model

### PR DESCRIPTION
**Pull request by paradime**
fixes https://github.com/PagerNation/PagerNation/issues/60

### CODE REVIEW

File: server/models/device.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const DeviceSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true
+  },
+  type: {
+    type: String,
+    required: true,
+    enum: ['email', 'phone', 'sms', 'user']
```
Comment (paradime) [inline server/models/device.js]: I was confused as to what you meant by `user` here. Can you elaborate @jarrydlee ?

Comment (kalebdavis) [inline server/models/device.js]: What does 'user' mean here?

File: server/models/device.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  address: {
```
Comment (kalebdavis) [inline server/models/device.js]: I'm assuming address means email address, phone number, etc. If that's the case, I don't think address is the best descriptor.

File: server/tests/models/device_spec.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import chai, { expect } from 'chai';
```
Comment (kalebdavis) [inline server/tests/models/device_spec.js]: This file should be called device.test.js to stay consistent with the other tests (ticket_spec should also be changed)

File: server/tests/models/device_spec.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        expect(err).to.not.exist;
+
+        expect(d.name).to.equal(device.name);
+        expect(d.type).to.equal(device.type);
+        expect(d.address).to.equal(device.address);
+        done();
+      });
+    });
+  });
+  context('when the type is not an enum', () => {
```
Comment (kalebdavis) [inline server/tests/models/device_spec.js]: Can you fix the whitespace in this file?

The tests are solid 👍

Comment (paradime) [inline server/tests/models/device_spec.js]: Can you clarify as to what you mean for 'whitespace'?

File: server/models/device.js (hunk 1 above)
Comment (paradime) [inline server/models/device.js]: I was curious as well @jarrydlee

Comment (jarrydlee) [inline server/models/device.js]: Yeah there were some changes that need to be made, `user` doesn't fit with our design

File: server/tests/models/device_spec.js (hunk 4 above)
Comment (kalebdavis) [inline server/tests/models/device_spec.js]: After some of these contexts there's a new line, and after others there isn't. I'd prefer a new line after each context, but whichever makes it consistent is fine.

File: server/models/device.js (hunk 2 above)
Comment (kalebdavis) [inline server/models/device.js]: `contact_information`?

File: server/tests/models/device.test.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import chai, { expect } from 'chai';
+import Device from '../../models/device';
+
+chai.config.incldueStack = true;
```
Comment (jarrydlee) [inline server/tests/models/device.test.js]: Typo

**Comment by kalebdavis** [comment]
Cool, looks good 👍 Can you please rebase, and put the ticket_spec change in its own commit? Technically it doesn't really go with this pull request, but having it in its own commit is good enough for me. Also, make sure your commit messages are in the imperative tense: "Remove user device" instead of "removed user device", etc.

**Pull request merged**